content/core-features/presence-and-occupancy/topic-presence_ia-update.textile

```diff
@@ -0,0 +1,737 @@
+      NSLog(@"%lu presence events received in 2nd page", [nextPage.items count]);
+    }];
+  }
+}];
+```
+
+```[swift]
+channel.presence.history { eventsPage, error in
+  let eventsPage = eventsPage!
+  print("\(eventsPage.items.count) presence events received in first page")
+  if eventsPage.hasNext {
+    eventsPage.next { nextPage, error in
+      print("\(nextPage!.items.count) presence events received in 2nd page")
+    }
+  }
+}
+```
+
+```[go]
+page, err := channel.Presence.History(nil)
+fmt.Println("%d messages in first page\n", len(page.PresenceMessages()))
+if page.hasNext() {
+  page2, err := page.Next()
+  fmt.Println("%d messages on 2nd page!\n", len(page2.PresenceMessages()))
+}
+```
+
+See the "presence history documentation":/api/rest-sdk/history#presence-history for further details of the supported query parameters.
+
+h3(#presence-multiple-client-id). Managing multiple client IDs
+
+h4(#realtime-presence-multiple-client-id). Realtime API
+
+p(tip). Each unique <span lang="default">"@clientId@":/api/realtime-sdk#client-id</span><span lang="ruby">"@client_id@":/api/realtime-sdk#client-id</span><span lang="csharp">"@ClientId@":/api/realtime-sdk#client-id</span> may only be present once when entering on behalf of another client as the unique identifier for each member in a presence set is the combined <span lang="default">"@clientId@":/api/realtime-sdk#client-id</span><span lang="ruby">"@client_id@":/api/realtime-sdk#client-id</span><span lang="csharp">"@ClientId@":/api/realtime-sdk#client-id</span> and shared "<span lang="default">@connectionId@</span><span lang="csharp">@ConnectionId@</span><span lang="ruby">@connection_id@</span>":/api/realtime-sdk/connection#id</span>
+
+An Ably client instance might, if on an application server for example, publish messages and be present on channels on behalf of multiple distinct "client IDs":/api/realtime-sdk#client-id. The channel's @Presence@ object therefore also supports methods that enable presence messages to be emitted for a <span lang="default">"@clientId@":/api/realtime-sdk#client-id</span><span lang="ruby">"@client_id@":/api/realtime-sdk#client-id</span><span lang="csharp">"@ClientId@":/api/realtime-sdk#client-id</span> specified at the time of the call, rather than implicitly based on the <span lang="default">"@clientId@":/api/realtime-sdk#client-id</span><span lang="ruby">"@client_id@":/api/realtime-sdk#client-id</span><span lang="csharp">"@ClientId@":/api/realtime-sdk#client-id</span> "specified when the library is instantiated or authenticated":/api/realtime-sdk#client-id.
+
+In order to be able to publish presence changes for arbitrary client IDs, the client library must have been instantiated either with an "API key":https://faqs.ably.com/what-is-an-app-api-key, or with a "token bound to a wildcard client ID":https://faqs.ably.com/can-a-client-emulate-any-client-id-i.e.-authenticate-using-a-wildcard-client-id.
+
+```[jsall](code-editor:realtime/presence-multiple-client-id)
```

Comment:
Check with @m-hulbert, but I think we should be removing jsbin references? If not, I'd make a note of these, as they will need to be removed at some point. You only need to remove the `(code-editor:realtime/presence-multiple-client-id)` bit.